towhee/dataframe/dataframe_v2.py

```diff
@@ -13,207 +13,447 @@
+from typing import List, Tuple, Any
 
-from towhee.dataframe.array import Array
+from towhee.dataframe.array.array import Array
+from towhee.dataframe._schema import _Schema
+from towhee.types._frame import _Frame
 
 
 class DataFrame:
     """
     A `DataFrame` is a collection of immutable, potentially heterogeneous blogs of data.
 
     Args:
+        columns (`list[Tuple[str, Any]]`)
+            The list of the column names and their corresponding data types.
         name (`str`):
             Name of the dataframe; `DataFrame` names should be the same as its
             representation.
-        data (`list[towhee.Array]` or `list[Tuple]` or `dict[str, towhee.Array]`):
+        data (`list[towhee.dataframe.Array]` or `list[Tuple]` or `dict[str, towhee.dataframe.Array]`):
             The data of the `DataFrame`. Internally, the data will be organized
             in a column-based manner.
     """
 
     def __init__(
         self,
-        name: str = None,
-        data=None,
-        columns=None,
+        columns: List[Tuple[str, str]],
+        name: str,
     ):
         self._name = name
-        self._len = 0
         self._sealed = False
-        self._lock = threading.Lock()
-
-        # For `data` is empty
-        if not data:
-            pass
-
-        # For `data` is `list`
-        elif isinstance(data, list):
-            container_types = set(type(i) for i in data)
-            if len(container_types) != 1:
-                raise ValueError(
-                    'can not construct Dataframe from a list of hybrid data containers. Try list[Tuple] or list[Array].')
-            container_type = container_types.pop()
-
-            # For `data` is `list[tuple]`
-            if container_type is tuple:
-                self._from_tuples(data, columns)
-            # For `data` is `list[towhee.Array]`
-            elif container_type is Array:
-                self._from_arrays(data, columns)
-            else:
-                raise ValueError('can not construct DataFrame from list[%s]' % (container_type))
 
-        # For `data` is `dict`
-        elif isinstance(data, dict):
-            self._from_dict(data)
+        self._iterator_lock = threading.RLock()
+        self._it_id = 0
+        self._iterators = {}
+        self._min_offset = 0
 
-        # Unrecognized data types
-        else:
-            raise ValueError('can not construct DataFrame from data type %s' % (type(data)))
+        self._block_lock = threading.RLock()
+        self._map_blocked = {}
+        self._window_blocked = {}
+
+
+        self._data_lock = threading.RLock()
+        self._len = 0
+        self._data_as_list = []
+        self._schema = _Schema()
+
+        self._initialize_storage(columns)
+
+    def _initialize_storage(self, columns):
+        """Set the columns in the schema."""
+        for name, tipe in columns:
+            self._schema.add_col(name=name, col_type = tipe)
+        if self._schema.col_index('_frame') is None:
```

Comment:
I think there is always be None, so the if is not needed.

Reply:
What about the situation where we are passing in data with _frame data already and just updating that row_id. Or should we not even include it in the schema and just use data[-1] when accessing frames.